fuse/readonly/readonly_unix.go

```diff
@@ -257,13 +257,15 @@ func (s *Node) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.ReadR
 	if err != nil {
 		return err
 	}
-
-	buf := resp.Data[:min(req.Size, int(int64(r.Size())-req.Offset))]
+	// Data has a capacity of Size
+	buf := resp.Data[:int(req.Size)]
```

Comment:
What is the out of bounds error that this is fixing? If the issue was that `buf` could be empty then we still get the error. 

Have we introduced another possible out of bounds error here if the response data is less than the request size? Is it really safe to remove the min check here?

Reply:
The fuse library gives us a `Data` buffer a capacity of `Size. I believe the issue was that `Size()` reported an incorrect size (-1, 0, etc.).

Reply:
I could be wrong here, but I don't think the issue was calculating `buf := resp.Data[-1]` because that would throw an `invalid slice index` instead of the `index out of range` error that was reported in the issue. 

Is it possible that `req.Size` is bigger than the buffer and so we'll run into an out of bounds error with this change?

Reply:
So, it _should_ be impossible: 

https://github.com/bazil/fuse/blob/7b5117fecadc6b83a347f794d6b8c4bbd0ece881/fs/serve.go#L1185-L1249

Reply:
But you're right, -1 would give us a different error. The only sane answer is that `Size` is too big.